### PR TITLE
Unique glove fibers

### PR DESCRIPTION
**Pull request by WarMechanic**
## About the PR

Every pair of gloves now has its own fingerprint, so items can be traced back to gloves which can then be traced back to people.

## Why / Balance

Evidence is very important to court cases running smoothly, so detectives now have more evidence that can certify whether or not John Syndicate's behaviour is valid. Traitors are now encouraged to either clean evidence off syndicate gear regardless of glove status, or use a disposable pair of gloves specifically for handling syndicate gear to be disposed of later.

Aside from being required to obfuscate evidence you leave behind, there is now a value proposition to searching glove prints of departments. Wearing gloves that does not correspond your department can punish an unknowing detective into searching the wrong people.

## Technical details

`FiberComponent.cs` now stores a Fiberprint variable like `FingerprintComponent.cs`. The code for assigning a fiberprint is the same as the fingerprint. When evidence is placed on an object, the fiberprint is concatenated to its localised fiber type.

## Media

hm ok we have these specific gloves on an akms
image
hm well we found the gloves and they have fingerprints
image
gotem
image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by DangerRevolution** [comment]
Test fail doesn't seem skbidi

**Comment by noctyrnal** [comment]
no more will yellow insulative fibres be the end of a case

File: Content.Server/Forensics/Components/FiberComponent.cs [hunk 1]
```diff
@@ -12,5 +12,8 @@ public sealed partial class FiberComponent : Component
 
         [DataField]
         public string? FiberColor;
+
+        [DataField]
+        public string? Fiberprint;
```
Comment (deltanedas) [inline Content.Server/Forensics/Components/FiberComponent.cs]: deltav comment

File: Content.Server/Forensics/Systems/ForensicsSystem.cs [hunk 2]
```diff
@@ -43,6 +44,10 @@ private void OnInteract(EntityUid uid, FingerprintComponent component, ContactIn
             ApplyEvidence(uid, args.Other);
         }
 
+        private void OnFiberInit(EntityUid uid, FiberComponent component, MapInitEvent args)
```
Comment (deltanedas) [inline Content.Server/Forensics/Systems/ForensicsSystem.cs:48]: deltav comment

File: Content.Server/Forensics/Systems/ForensicsSystem.cs [hunk 3]
```diff
@@ -202,9 +207,9 @@ private void OnCleanForensicsDoAfter(EntityUid uid, ForensicsComponent component
                 targetComp.Residues.Add(string.IsNullOrEmpty(residue.ResidueColor) ? Loc.GetString("forensic-residue", ("adjective", residue.ResidueAdjective)) : Loc.GetString("forensic-residue-colored", ("color", residue.ResidueColor), ("adjective", residue.ResidueAdjective)));
         }
 
-        public string GenerateFingerprint()
+        public string GenerateFingerprint(int length = 16)
```
Comment (deltanedas) [inline Content.Server/Forensics/Systems/ForensicsSystem.cs]: .

File: Content.Server/Forensics/Systems/ForensicsSystem.cs [hunk 4]
```diff
@@ -231,7 +236,12 @@ private void ApplyEvidence(EntityUid user, EntityUid target)
             if (_inventory.TryGetSlotEntity(user, "gloves", out var gloves))
             {
                 if (TryComp<FiberComponent>(gloves, out var fiber) && !string.IsNullOrEmpty(fiber.FiberMaterial))
-                    component.Fibers.Add(string.IsNullOrEmpty(fiber.FiberColor) ? Loc.GetString("forensic-fibers", ("material", fiber.FiberMaterial)) : Loc.GetString("forensic-fibers-colored", ("color", fiber.FiberColor), ("material", fiber.FiberMaterial)));
+                {
+                    var fiberLocale = string.IsNullOrEmpty(fiber.FiberColor)
```
Comment (deltanedas) [inline Content.Server/Forensics/Systems/ForensicsSystem.cs:244]: .

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by WarMechanic** [comment]
i forgor the comments

**Review by deltanedas — DISMISSED** [review]

**Comment by deltanedas** [comment]
moths keep winning

**Comment by WarMechanic** [comment]
did i just fuck it up

**Review by Axionyxx — APPROVED** [review]
Reapproving because only a whitespace got added

**Comment by WarMechanic** [comment]
thank you null

**Pull request merged**